src/mosaic.mjs

```diff
@@ -87,14 +87,66 @@ function requestCachedMosaic512px(z, x, y) {
   return request;
 }
 
+async function bufferToPixels(buffer) {
+
+  if (!buffer || buffer.length === 0 || buffer === undefined) {
+    console.log('Input buffer is empty or undefined', !buffer, buffer.length);
+    return null;
+  }
+
+  const { data, info } = await sharp(buffer)
+    .raw()
+    .toBuffer({ resolveWithObject: true });
+
+  const pixels = [];
+  for (let i = 0; i < data.length; i += 4) {
+    pixels.push({
+      r: data[i],
+      g: data[i + 1],
+      b: data[i + 2],
+      a: data[i + 3]
+    });
+  }
+
+  return { pixels, width: info.width, height: info.height };
+}
+
+async function pixelsToBuffer(pixels, width, height) {
+  if (!pixels || pixels.length === 0 || pixels === undefined) {
+    console.log('Input pixels are empty or undefined');
+    return null;
+  }
+
+  const data = new Uint8Array(width * height * 4); // Keeping with RGBA format from the PNG
+  for (let i = 0; i < pixels.length; i++) {
+    const blackThreshold = 33;
```

Comment:
There are also white nodata regions, not only black. 

The idea is to choose the best pixel among all available, and not just clip near-black from the source images. That would give some wiggle space to handle other issues like multiresolution composite and clouds/haze.